python/az/aro/azext_aro/tests/latest/unit/test_validators.py

```diff
@@ -0,0 +1,419 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the Apache License 2.0.
+
+from typing import List
+from unittest.mock import Mock, patch
+from azext_aro._validators import (
+    validate_cidr, validate_client_id, validate_client_secret, validate_cluster_resource_group,
+    validate_disk_encryption_set, validate_domain, validate_pull_secret, validate_sdn, validate_subnet, validate_subnets,
+    validate_visibility, validate_vnet_resource_group_name, validate_worker_count, validate_worker_vm_disk_size_gb, validate_refresh_cluster_credentials
+)
+from azure.cli.core.azclierror import (
+    InvalidArgumentValueError, RequiredArgumentMissingError, RequiredArgumentMissingError, CLIInternalError
+)
+from azure.core.exceptions import ResourceNotFoundError
+import pytest
+
+
+test_validate_cidr_data = [
+    ("should not raise exception when valid IPv4 address", Mock(key='192.168.0.0/28'), "key", None),
+    ("should raise InvalidArgumentValueError when non valid IPv4 address due to beeing a simple string", Mock(key='this is an invalid network'), "key", InvalidArgumentValueError),
+    ("should raise InvalidArgumentValueError when non valid IPv4 address due to invalid network ID", Mock(key='192.168.0.0.0.0/28'), "key", InvalidArgumentValueError),
+    ("should raise InvalidArgumentValueError when non valid IPv4 address due to invalid hostID", Mock(key='192.168.0.0.0.0/2888'), "key", InvalidArgumentValueError),
```

Comment:
I'm having trouble differentiating between these tests. Both contain invalid IP address CIDRs due to too many IP digits `192.168.0.0.0.0`, but the second also contains an invalid range of `2888`

Reply:
Thanks, well spotted. Test case on line 22 should fail just due to an invalid range of 288, fixed.